transport/socks5/packet_listener_test.go

```diff
@@ -0,0 +1,107 @@
+package socks5
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/things-go/go-socks5"
+)
+
+func TestSOCKS5Associate(t *testing.T) {
+	// Create a local listener
+	// This creates a UDP server that responded to "ping"
+	// message with "pong" response
+	locIP := net.ParseIP("127.0.0.1")
+	// Create a local listener
+	echoServerAddr := &net.UDPAddr{IP: locIP, Port: 12199}
+	echoServer := setupUDPEchoServer(t, echoServerAddr)
+	defer echoServer.Close()
+
+	// Create a socks server to proxy "ping" message
+	cator := socks5.UserPassAuthenticator{Credentials: socks5.StaticCredentials{
+		"testusername": "testpassword",
+	}}
+	proxySrv := socks5.NewServer(
+		socks5.WithAuthMethods([]socks5.Authenticator{cator}),
+		//socks5.WithLogger(socks5.NewLogger(log.New(os.Stdout, "socks5: ", log.LstdFlags))),
+	)
+	// Start listening
+	proxyServerAddress := "127.0.0.1:12355"
+	go func() {
+		err := proxySrv.ListenAndServe("tcp", proxyServerAddress)
+		require.NoError(t, err)
+	}()
+	time.Sleep(10 * time.Millisecond)
+
+	// Connect, auth and connec to local server
```

Comment:
It's misspelled, and the local server is the proxy?


Perhaps:
```suggestion
	// Connect to local proxy, auth and start the PacketConn.
```